pkg/yurtmanager/controller/autonomy/autonomy_controller.go

```diff
@@ -0,0 +1,135 @@
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an AS IS BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package autonomy
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	appconfig "github.com/openyurtio/openyurt/cmd/yurt-manager/app/config"
+	"github.com/openyurtio/openyurt/cmd/yurt-manager/names"
+	appsv1beta1 "github.com/openyurtio/openyurt/pkg/apis/apps/v1beta1"
+	"github.com/openyurtio/openyurt/pkg/projectinfo"
+)
+
+var (
+	controllerResource = v1.SchemeGroupVersion.WithKind("Node")
+)
+
+type ReconcileAutonomy struct {
+	client.Client
+}
+
+func Format(format string, args ...interface{}) string {
+	s := fmt.Sprintf(format, args...)
+	return fmt.Sprintf("%s: %s", names.AutonomyController, s)
+}
+
+func Add(ctx context.Context, c *appconfig.CompletedConfig, mgr manager.Manager) error {
+	klog.Infof(Format("autonomy-controller add controller %s", controllerResource.String()))
+	return add(mgr, c, newReconciler(c, mgr))
+}
+
+func newReconciler(_ *appconfig.CompletedConfig, mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileAutonomy{Client: mgr.GetClient()}
+}
+
+func add(mgr manager.Manager, cfg *appconfig.CompletedConfig, r reconcile.Reconciler) error {
+	c, err := controller.New(names.AutonomyController, mgr, controller.Options{
+		Reconciler:              r,
+		MaxConcurrentReconciles: int(cfg.Config.ComponentConfig.AutonomyController.ConcurrentAutonomyWorkers),
+	})
+	if err != nil {
+		return err
+	}
+
+	nodePredicate, nodeMapFunc := prepare()
+	err = c.Watch(&source.Kind{Type: &v1.Node{}}, handler.EnqueueRequestsFromMapFunc(nodeMapFunc), nodePredicate)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *ReconcileAutonomy) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	klog.V(2).Infof("Reconcile Node %s/%s Start.", req.Namespace, req.Name)
+	var node v1.Node
+	err := r.Client.Get(ctx, req.NamespacedName, &node)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == appsv1beta1.NodeAutonomy && condition.Status == v1.ConditionTrue {
+			if node.Labels == nil {
+				node.Labels = make(map[string]string)
+			}
+			node.Labels[projectinfo.GetAutonomyStatusLabel()] = "true"
```

Comment:
GetAutonomyStatusLabel is not located in node annotations?

Reply:
by the way, we also need to change `GetAutonomyStatusLabel ` to false when node condition is false.

Reply:
GetAutonomyAnnotation is in node annotations and it is used by users to open node autonomy. And the real autonomy status is reflected in the label `GetAutonomyStatusLabel`.